subprojects/base-services/src/main/java/org/gradle/internal/hash/FileHasher.java

```diff
@@ -24,7 +24,7 @@
 
 public interface FileHasher {
     /**
-     * Returns the hash of the given input stream.
+     * Returns the hash of the given input stream. This method buffers reads on the stream.
```

Comment:
I don't understand why we have to go black-and-white here. Why does it have to be mentioned nowhere or everywhere? Do you really intend to make one big PR hunting down all the spots?

I think having the information in some places is better than having it nowhere and encourages us to add it in more places going forward. 

That said, if you insist I can remove this line.